src/komodo_utils.h

```diff
@@ -1759,7 +1759,7 @@ void komodo_args(char *argv0)
 
         for ( int i = 0; i < ASSETCHAINS_MAX_ERAS; i++ )
         {
-            if ( ASSETCHAINS_DECAY[i] == 100000000 && ASSETCHAINS_ENDSUBSIDY == 0 )
+            if ( ASSETCHAINS_DECAY[i] == 100000000 && ASSETCHAINS_ENDSUBSIDY[i] == 0 )
```

Comment:
this may affect existing chains. I have checked known chain and found only one param set that may be affected (https://dexstats.info/params.php):
`./verusd -ac_name=VRSC -ac_algo=verushash -ac_cc=1 -ac_veruspos=50 -ac_supply=0 -ac_eras=3 -ac_reward=0,38400000000,2400000000 -ac_halving=1,43200,1051920 -ac_decay=100000000,0,0 -ac_end=10080,226080,0 -ac_timelockgte=19200000000 -ac_timeunlockfrom=129600 -ac_timeunlockto=1180800 -addnode=185.25.48.236 -addnode=185.64.105.111`
(luckily ac_end not zero for -ac_decay=100000000 so this fix is okay)

another chain:
`./komodod -ac_name=LABS -ac_supply=350689 -ac_reward=0,0,800000000 -ac_decay=0,100000000,100000000 -ac_halving=129,1 -ac_end=128,10080,0 -ac_notarypay=64,100000000,1000000000 -ac_eras=3 -ac_staked=51 -ac_sapling=1 -ac_cc=101 -ac_ccenable=226,236 -ac_cclib=labs -addnode=80.240.17.222 -addnode=195.201.20.230`
this chain may be affected but seems this chain is not used already

Reply:
Should I put an extra check here? Perhaps ... `&& (ASSETCHAINS_ENDSUBSIDY == 0 || ASSETCHAINS_ENDSUBSIDY[i] == 0)` or some storage/check of the length of valid elements in the array?